### PR TITLE
IGNITE-19652 Adapt to DirectByteBuffer ctor changes in JDK 21

### PR DESCRIPTION
**Pull request by wendigo**
Signature has changed in https://github.com/openjdk/jdk/commit/a56598f5a534cc9223367e7faa8433ea38661db9

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/util/GridUnsafe.java [hunk 1]
```diff
@@ -1660,6 +1661,12 @@ private static Method newDirectBufferMethod(Object nioAccessObj) {
         return javaVer < 9 ? "sun" : "jdk.internal";
     }
 
+    @NotNull private static boolean directByteBufferLongLongCtor() {
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/internal/util/GridUnsafe.java:1664]: I suggest to declare a local var in static class initialization and pass it to methods:

```
boolean isDirectBufLongCap = majorJavaVersion(jdkVersion()) >= 21;
```

**Comment by NSAmelchev** [comment]
Hello, @wendigo.
Please, fix the codestyle.
I suggest writing to the dev list to ask for help with running CI on a new JDK.

**Comment by wendigo** [comment]
@NSAmelchev ping

**Review by ptupitsyn — APPROVED** [review]

**Comment by wendigo** [comment]
@ptupitsyn do you happen to know when the next minor release is happening?

**Comment by ptupitsyn** [comment]
@wendigo there is no set date, but I would expect it by the end of the year.

**Comment by wendigo** [comment]
@ptupitsyn that’s too late for us in Trino. I guess that I’ll need to patch it somehow

**Pull request merged**